File: CompilationTest/StructureTest1/results/StructureTest1.ts
class MyList<T> {
    items: OneArray;
}

class Item {
    offset: number = 5;
    strTest: string = "test" + "test2";
    strConstr: string = "constr";

    constructor(strConstr: string) {
        this.strConstr = strConstr;
    }
}

class Container {
    itemList: MyList<Item>;
    stringList: MyList<string>;

    method0() {
    }
  
    method1(str: string) {
        return str;
    }
}

console.log("ok");

File: CompilationTest/StructureTest1/results/StructureTest1.go
package main

import "fmt"

type MyList struct {
    Items []interface{}
}

func NewMyList() *MyList {
    this := new(MyList)
    return this
}

type Item struct {
    Offset int
    StrTest string
    StrConstr string
}

func NewItem(strConstr string) *Item {
    this := new(Item)
    this.Offset = 5
    this.StrTest = "test" + "test2"
    this.StrConstr = "constr"
    this.StrConstr = strConstr
    return this
}

type Container struct {
    ItemList *MyList<Item>
    StringList *MyList<string>
}

func NewContainer() *Container {
    this := new(Container)
    return this
}

func (this *Container) Method0() {
}

func (this *Container) Method1(str string) string {
    return str
}

func main() {
    fmt.Println("ok")
}